### ISSUE TITLE
TF 2.3.0 build failure when its conda package is built using Anaconda's toolchain 

### ISSUE THREAD

**Issue by npanpaliya** [tensorflow/tensorflow#42313]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux RHEL 7 ppc64le
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): Source v2.3.0
- TensorFlow version: 2.3.0
- Python version: 3.6/3.7
- Installed using virtualenv? pip? conda?: Building conda package of TF 2.3
- Bazel version (if compiling from source): 3.1.0
- GCC/Compiler version (if compiling from source): 7.3
- CUDA/cuDNN version: 10.2/7

**Describe the problem**
I'm trying to build conda package of TF 2.3 using Anaconda's toolchain with gcc 7.3 on ppc64le. And with gpu variant build, I'm getting following error -

```
ERROR: /home/builder/.cache/bazel/_bazel_builder/738bada2dae39de8837b3ef244ba3e1e/external/flatbuffers/BUILD.bazel:64:1: Linking of rule '@flatbuffers//:flatc' failed (Exit 1): crosstool_wrapper_driver_is_not_gcc failed: error executing command
  (cd /home/builder/.cache/bazel/_bazel_builder/738bada2dae39de8837b3ef244ba3e1e/execroot/org_tensorflow && \
  exec env - \
    CUDA_TOOLKIT_PATH=/opt/anaconda3/conda-bld/tensorflow-base_1597311086398/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla \
    GCC_HOST_COMPILER_PATH=/opt/anaconda3/conda-bld/tensorflow-base_1597311086398/_build_env/bin/powerpc64le-conda_cos7-linux-gnu-cc \
    LD_LIBRARY_PATH=::/opt/anaconda3/conda-bld/tensorflow-base_1597311086398/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/extras/CUPTI/lib64::/opt/anaconda3/conda-bld/tensorflow-base_1597311086398/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/cuda/lib: \
    PATH=/opt/anaconda3/conda-bld/tensorflow-base_1597311086398/_build_env/bin:/opt/anaconda3/conda-bld/tensorflow-base_1597311086398/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/bin:/opt/anaconda3/condabin:/opt/anaconda3/conda-bld/tensorflow-base_1597311086398/_build_env:/opt/anaconda3/conda-bld/tensorflow-base_1597311086398/_build_env/bin:/opt/anaconda3/conda-bld/tensorflow-base_1597311086398/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla:/opt/anaconda3/conda-bld/tensorflow-base_1597311086398/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/bin:/opt/anaconda3/bin:/opt/anaconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin \
    PWD=/proc/self/cwd \
    PYTHON_BIN_PATH=/opt/anaconda3/conda-bld/tensorflow-base_1597311086398/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/bin/python \
    PYTHON_LIB_PATH=/opt/anaconda3/conda-bld/tensorflow-base_1597311086398/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/lib/python3.6/site-packages \
    TF2_BEHAVIOR=1 \
    TF_CONFIGURE_IOS=0 \
    TF_CUDA_COMPUTE_CAPABILITIES=3.7,6.0,7.0,7.5 \
    TF_CUDA_PATHS=/opt/anaconda3/conda-bld/tensorflow-base_1597311086398/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla \
    TF_CUDA_VERSION=10.2 \
    TF_CUDNN_VERSION=7 \
    TF_ENABLE_XLA=1 \
    TF_NCCL_VERSION=2 \
    TF_NEED_CUDA=1 \
    TF_SYSTEM_LIBS=org_sqlite \
  external/local_config_cuda/crosstool/clang/bin/crosstool_wrapper_driver_is_not_gcc @bazel-out/ppc-opt/bin/external/flatbuffers/flatc-2.params)
Execution platform: @local_execution_config_platform//:platform
powerpc64le-conda_cos7-linux-gnu-cc: error: bazel-out/ppc-opt/bin/external/flatbuffers/flatc: No such file or directory
Target @flatbuffers//:flatc failed to build
```

Even on Linux x86_64 with gcc 5.4 I'm seeing similar issue with some or other shared library being generated by build. Also, note that only GPU build is getting affected. CPU builds are working fine. And this error occurs only within conda environment. Standalone build of TF using gcc 7.4 works fine on ppc64le.

**Any other info / logs**
When I digged into the TF commits that went into 2.3.0, I found the change done in this commit https://github.com/tensorflow/tensorflow/commit/e0b19f6ef223af40e2e6d1d21b8464c1b2ebee8f#diff-ee64a9d8638aa0a655dfaac625f72bb2 has been causing the above problem. Since the exact problem is not mentioned in the bazel output, I am unable to find out what has gone wrong in the above commit. Reverting the change done in this cc_toolchain_config.bzl.tpl file fixes the problem for me.

If somebody could help me figure out what (specifically from above commit) might caused this error, it would be really appreciated.

**Comment by ravikyram** [comment]
@npanpaliya 

Sorry, but we don't provide support for issues with the conda environment.
This issue is more suitable on Continuum [Anaconda repo](https://github.com/ContinuumIO/anaconda-issues/issues) since its related to TF installation with Anaconda.
Please post it on [Continuum Anaconda](https://github.com/ContinuumIO/anaconda-issues/issues).
Thanks!

**Comment by npanpaliya** [comment]
@ravikyram Thanks for your response. I'll raise the issue with Anaconda too.

I raised the issue here because conda builds used to work before that particular commit and then it has started giving error due to changes done in cc_toolchain_config.bzl.tpl file. So, just want to know if anyone has any pointers on it. The refactoring done in that file is for sure the cause of this error.

**Comment by npanpaliya** [comment]
I've further figured out the exact linker related changes that are causing the error. Changes done in https://github.com/tensorflow/tensorflow/commit/bb4c751414c3562ab3ab4298f866f47438078c37 have introduced the build issues in conda environment which otherwise work fine.

**Comment by jvishnuvardhan** [comment]
@npanpaliya One thing I noted was that you are using CUDA10.2 where as tested build configuration of TF2.3 is as follows.

Version | Python version | Compiler | Build tools | cuDNN | CUDA
-- | -- | -- | -- | -- | --
tensorflow-2.3.0 | 3.5-3.8 | GCC 7.3.1 | Bazel 3.1.0 | 7.6 | 10.1

May be there is a in compatibility issue. I am not sure how TF-Anaconda package was built but pip binaries of TF were built with the above config. Thanks

**Comment by npanpaliya** [comment]
Thanks @jvishnuvardhan . But we have been building conda package for TF 2.2 too with cuda 10.2 and even TF 2.3 works with cuda 10.2 if I revert the commit I mentioned in my above comment. The build error I'm seeing is due to linker options' ordering changes done in that commit.

**Comment by jvishnuvardhan** [comment]
@npanpaliya Unfortunately we don't support or maintain Conda packages. I believe you need to post this issue is Anaconda repository. Thanks!

**Comment by npanpaliya** [comment]
I've already posted this issue in Anaconda repository. 
Also, tried building TF 2.3 conda package with cuda 10.1 but the error persists. So, cuda version doesn't seem to be problematic.

**Comment by gunan** [comment]
I have only reviewed the said change, but I have no knowledge of the conda toolchain.
@hlopko could you take a look if you have time?

**Comment by njzjz** [comment]
@npanpaliya https://github.com/tensorflow/tensorflow/issues/41856#issuecomment-706422403 may be helpful for you.

**Comment by npanpaliya** [comment]
Thanks @njzjz . I was able to get this working by reverting commit https://github.com/tensorflow/tensorflow/commit/bb4c751414c3562ab3ab4298f866f47438078c37 in my builds. Indeed what you've pointed is also on the same lines.

**Comment by Flamefire** [comment]
@npanpaliya You may also test my fix from #43951

**Comment by npanpaliya** [comment]
@Flamefire - This seems to be correct. I'd noticed `-B` to be alone in bazel's params file which contains linker options for the target but I think I overlooked it. I'll try this fix. Thanks a lot!
